test/DynamoCoreTests/UtilityTests.cs

```diff
@@ -589,6 +589,25 @@ public void MakeAbsolutePath04()
         }
 
         [Test]
+        [Category("UnitTests")]
+        public void FindFilesWithcaseinsensitiveFileExtensions()
+        {
+            var path = Path.GetTempPath();
+            string filePath = Path.Combine(path, "SaveFile.txt");
+            using (StreamWriter sw = new StreamWriter(filePath))
+            {
+                sw.Write("test");
+            }
+
+            Assert.AreEqual(filePath, PathHelper.FindFileInPaths("SaveFile", ".txt", new String[] { path }));
+            Assert.AreEqual(filePath, PathHelper.FindFileInPaths("SaveFile", ".TXT", new String[] { path }));
+            Assert.AreEqual(filePath, PathHelper.FindFileInPaths("SaveFile", ".tXt", new String[] { path }));
```

Comment:
This isn't testing what you want to test.
You want to write in turn `SaveFile.txt`, `SaveFile.TXT`, and `SaveFile.tXt`, and then make sure that you find them with FindFileInPaths(..., ".txt", ...);